packages/config/src/class-config.php

```diff
@@ -208,6 +240,20 @@ protected function ensure_options_connection() {
 			}
 
 			( new Plugin( $slug ) )->add( $name, $options );
+
+			$connection = new Manager( $slug );
+
+			if ( ! empty( $options['deactivate_disconnect'] ) ) {
+				if ( empty( $options['plugin_file'] ) ) {
```

Comment:
You can also grab caller information from the PHP runtime and use that to figure out the slug of the calling plugin